components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java

```diff
@@ -81,6 +84,14 @@ public void handleEvent(Event event) throws IdentityEventException {
                 IdentityEventConstants.Event.POST_SET_USER_CLAIMS.equals(eventName)) {
             enable = Boolean.parseBoolean(Utils.getConnectorConfig(IdentityRecoveryConstants.ConnectorConfig
                     .ENABLE_EMAIL_VERIFICATION_ON_UPDATE, user.getTenantDomain()));
+            if (!enable) {
+                /* We need to empty 'EMAIL_ADDRESS_PENDING_VALUE_CLAIM' because having a value in that claim implies
+                a verification is pending. But verification is not enabled anymore. */
+                if (claims.containsKey(IdentityRecoveryConstants.EMAIL_ADDRESS_CLAIM)) {
+                    invalidatePendingEmailVerification(user, userStoreManager, claims);
+                }
+                return;
```

Comment:
Let's remove the return; statement here. Otherwise, it will not reach this log message: https://github.com/wso2-extensions/identity-governance/pull/414/files#diff-5044674a0bf99578dc84a4b6fa7ebd2aR100

Reply:
Addressed in https://github.com/wso2-extensions/identity-governance/pull/414/commits/b5f0c45e2ce7cb40a24f7a051acc52455755cce4